client/src/pages/datasets/datasetContentsView/[datasetName].tsx

```diff
@@ -21,19 +28,123 @@ export default function DatasetViewer() {
   }, [user]);
 
   const getDatasetFiles = async () => {
-    console.log(datasetName);
-    const allFiles: UserFile[] = await getFilesFromDataset(user!, datasetName);
-    console.log(allFiles);
-    allFiles.forEach(file => {
-      console.log(file.fileName);
-    });
+    const dataset: Dataset = await getDataset(user!, datasetName!);
+    const allFiles: UserFile[] = await Promise.all(
+      dataset.files.map(filename => getFile(user!, filename))
+    );
+    setCurrentDataset(dataset);
     setCurrentDatasetFiles(allFiles);
+    setCurrentState('completed');
+  };
+
+  /**
+   * A dictionary-like object to map object properties to user friendly
+   * text in tables
+   */
+  const keysToStrings: {[key: string]: string} = {
+    punctuationToRemove: 'Punctuation to remove',
+    punctuationToReplace: 'Punctuation to replace',
+    tagsToRemove: 'Tags to remove',
+    wordsToRemove: 'Words to remove',
+    selectionMechanism: 'Selection Mechanism',
+    selectionValue: 'Selection Value',
+  };
+
+  /**
+   * Gets a list of JSX elements that represent the rows of the table containing
+   * DataPreparationOptions for the given dataset
+   *
+   * @param dataset Dataset whose Data-Preparation-options need to be obtained
+   * @returns A list of JSX elements
+   */
+  const getDataPreparationOptionsTableRows = (dataset: Dataset) => {
+    const dataPrepOptions = dataset.options;
+    let key: keyof typeof dataPrepOptions;
+    const tableRows = [];
+    for (key in dataPrepOptions) {
```

Comment:
Nit: prefer map and filter over the imperative approaches where possible.

Reply:
@harrykeightley I am actually not sure how to do this using map/filter because I am iterating over the properties of an interface. Maybe I am missing something obvious. Would you know how I can do it?